### ISSUE TITLE
Add more complicated data integrity checks to data validation

### ISSUE THREAD

**Issue by njarboe** [earthref/MagIC#263]
Add more complicated data integrity checks to data validation.

Here is a list I compiled from the old data console a few year ago: I have removed ones not longer valid with the new website and data model and made some edits.

List of extra checks the console does:

Check for duplicates in data in some tables: For columns that require a unique key, check for duplicates.

Check email addresses: For columns with header email check to see if email is in the form "*??@??*.??*"

Check url: If column header like "*_url*" then check if url is in the form "http*://??*.??*" or https

Check if doi: If column header = "doi" then check if doi like "10.*" Can we check if the doi resolves and give a warning. That would be great.

Check all controlled vocabulary fields: Make sure the values of a field that has a controlled vocabulary match a value in the controlled vocabulary.

**Comment by rminnett** [comment]
Yep, those are all possible, including checking if a doi or url resolve.
